### PR TITLE
Prefer AuthenticatedByShopify concern over AuthenticatedController

### PR DESCRIPTION
**Pull request by kdonovan**
I ran into the bug pointed out in #541, where the docs incorrectly claim `ShopifyApp::AuthenticatedController` inherits from `ApplicationController`. 

I understand the logic behind changing this to avoid accidentally importing breaking logic changes from the user's application, but it still causes issues as e.g. my local app has customizations to the ApplicationController to add support for Pundit and many other things. Any "userland" additions to Shopify integration will need _both_ my expected `ApplicationController` logic _and_ the default Shopify authentication logic; I'd originally worked around that by extracting all my `ApplicationController` code into a new `ApplicationControllerConcern` concern of my own, and injecting that into both the standard `ApplicationController` and a separate `Shopify::ApplicationController` that inherits from `ShopifyApp::AuthenticatedController` (and using that as the base for all my additional Shopify controllers). It works... but it's hacky and unpleasant, and I was inspired once I saw from #541 that this behavior wasn't exactly completely polished/decided upon.

After this PR, all my local workarounds become unnecessary. Userland Shopify controllers inherit from `ApplicationController`, like any other controller, and merely include the new concern:

```ruby
class SomeNewShopifyController < ApplicationController
  include ShopifyApp::AuthenticatedByShopify
end
```

The test suite is still green, and I can't see any obvious issues... but I'd appreciate another set of eyes to make sure I'm not missing anything obvious. 👀

### CODE REVIEW

**Comment by vfonic** [comment]
This also makes it consistent with [`AppProxyVerification`](https://github.com/Shopify/shopify_app/blob/master/lib/shopify_app/controller_concerns/app_proxy_verification.rb) and [`WebhookVerification`](https://github.com/Shopify/shopify_app/blob/master/lib/shopify_app/controller_concerns/webhook_verification.rb).

I would just change the name slightly (following the two above-mentioned concerns). Perhaps `ShopifyApp::AuthenticationVerification`?

File: lib/shopify_app/controller_concerns/authenticated_by_shopify.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+      include ShopifyApp::LoginProtection
+      include ShopifyApp::EmbeddedApp
+
+      protect_from_forgery with: :exception
+      before_action :login_again_if_different_shop
+      around_action :shopify_session
+    end
+
+  end
+end
```
Comment (kevinhughes27) [inline lib/shopify_app/controller_concerns/authenticated_by_shopify.rb]: can you add the newline char here please

**Comment by kdonovan** [comment]
Apologies on the delay here -- day job went a little haywire -- but I'll definitely circle back to this project shortly and I'll update the PR as discussed above then. 👍

**Comment by kdonovan** [comment]
@EiNSTeiN- Apologies again for the delay, but I was able to circle back to this today and update the generators + add a deprecation warning.

Note: I started looking at the failing code-review/policial checks, but implementing those (e.g. frozen string literal comment) in these files actually diverges the style from the existing codebase.

**Pull request closed without merging**